kong/resource_kong_plugin_test.go

```diff
@@ -158,6 +161,71 @@ func TestAccKongPluginImportConfigJson(t *testing.T) {
 	})
 }
 
+func TestAccKongPlugin_Upsert(t *testing.T) {
+
+	uniqueConstraintError, _ := regexp.Compile(".*unique constraint violation.*")
+
+	TestProvider_configure(t)
+
+	os.Unsetenv("KONG_UPSERT_RESOURCES")
+	defer os.Unsetenv("KONG_UPSERT_RESOURCES")
+
+	// Simulate that plugin already exists but with different values.
+	request := &gokong.PluginRequest{
+		Name: "hmac-auth",
+		Config: map[string]interface{}{
+			"algorithms": []string{
+				"hmac-sha384",
+				"hmac-sha512",
+			},
+		},
+	}
+	plugin, err := testAccProvider.Meta().(*config).adminClient.Plugins().Create(request)
+	if err != nil {
+		t.Fatalf("could not create plugin: %v", err)
+	}
+	expectedPluginID := plugin.Id
+
+	pluginConf := `
+resource "kong_plugin" "hmac_auth" {
+	name  = "hmac-auth"
+	config_json = <<EOT
+	{
+		"algorithms": [
+			"hmac-sha1",
+			"hmac-sha256"
+		]
+	}
+EOT
+}`
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			// With upsert disable (default value), this will raise a unique constraint error.
+			{
+				Config:      pluginConf,
```

Comment:
I'd add a `PreConfig` here as well and call `os.Unsetenv("KONG_UPSERT_RESOURCES")` to make the intention clearer.

It'd allow to remove it line 170